### PR TITLE
[Api][Admin] Enable Order resource

### PR DESCRIPTION
**Pull request by Wojdylak**
| Q               | A
|-----------------|-----
| Branch?         | api-platform-3
| Bug fix?        | no
| New feature?    | no
| BC breaks?      | no
| Deprecations?   | no
| Related tickets | n/a
| License         | MIT

### CODE REVIEW

File: src/Sylius/Bundle/ApiBundle/Resources/config/services/filters.xml [hunk 1]
```diff
@@ -132,54 +132,38 @@
...
-<!--            <tag name="api_platform.filter" />-->
-<!--        </service>-->
-
-<!--        <service id="sylius.api.order_variants_filter" parent="api_platform.doctrine.orm.search_filter" public="true">-->
-<!--            <argument type="collection">-->
-<!--                <argument key="items.variant.translations.name">exact</argument>-->
-<!--            </argument>-->
-<!--            <tag name="api_platform.filter" />-->
-<!--        </service>-->
+        <service id="sylius_api.admin.search_filter.order" parent="api_platform.doctrine.orm.search_filter" public="true">
```
Comment (NoResponseMate) [inline src/Sylius/Bundle/ApiBundle/Resources/config/services/filters.xml]:
```suggestion
        <service id="sylius_api.search_filter.admin.order" parent="api_platform.doctrine.orm.search_filter" public="true">
```
Up to now, every other section-based service id has the section after the type of service:
- `sylius_api.state_provider.admin.order.adjustment.collection`
- `sylius_api.state_processor.admin.promotion_coupon.remove`
- `sylius_api.doctrine.orm.query_extension.shop.address.shop_user_based`

etc.

File: src/Sylius/Bundle/ApiBundle/Resources/config/services/filters.xml [hunk 2]
```diff
@@ -132,54 +132,38 @@
...
+            <argument type="collection">
+                <argument key="channel.code" />
+                <argument key="currencyCode">exact</argument>
+                <argument key="customer.id">exact</argument>
+                <argument key="items.productName">exact</argument>
+                <argument key="shipments.method.code">exact</argument>
+                <argument key="items.variant.translations.name">exact</argument>
```
Comment (NoResponseMate) [inline src/Sylius/Bundle/ApiBundle/Resources/config/services/filters.xml:142]: Not related; why do we check the current variant name, but a potentially snapshotted product name?

Comment (Wojdylak) [reply src/Sylius/Bundle/ApiBundle/Resources/config/services/filters.xml:142]: I don't know

File: src/Sylius/Bundle/ApiBundle/Attribute/OrderTokenValueAware.php [hunk 3]
```diff
@@ -0,0 +1,22 @@
...
+ */
+
+declare(strict_types=1);
+
+namespace Sylius\Bundle\ApiBundle\Attribute;
+
+#[\Attribute(\Attribute::TARGET_CLASS)]
+final class OrderTokenValueAware
+{
+    public function __construct(public string $constructorArgumentName = 'orderTokenValue')
```
Comment (NoResponseMate) [inline src/Sylius/Bundle/ApiBundle/Attribute/OrderTokenValueAware.php:19]: Maybe we should extract the default names to consts within the attributes (or better yet, the interface) and reuse that in here and in the context builders;
Right now we have the same data hardcoded in multiple places, and the relation between the interface, attribute and context builder is muddled.

But that's something for another PR.

**Review by NoResponseMate — APPROVED** [review]

**Comment by NoResponseMate** [comment]
Thank you, @Wojdylak!

**Pull request merged**